components/uuid_vx_udf/CMakeLists.txt

```diff
@@ -33,6 +33,19 @@ MYSQL_ADD_COMPONENT(uuid_vx_udf
 
 target_include_directories(component_uuid_vx_udf SYSTEM PRIVATE ${BOOST_PATCHES_DIR} ${BOOST_INCLUDE_DIR})
 
-if(NOT APPLE)
-  target_link_libraries(component_uuid_vx_udf PRIVATE atomic)
+# Check if -latomic is required or not
```

Comment:
@lukin-oleksiy, I like this approach for detecting the same from `protobuf` more.
https://github.com/percona/percona-server/blob/646a7cf4b6874eb4a9163078febc13c7938328e1/extra/protobuf/protobuf-24.4/CMakeLists.txt#L184
1. They include `CheckCXXSourceCompiles` CMake include for `CHECK_CXX_SOURCE_COMPILES`
   `include(CheckCXXSourceCompiles)`
2. They save and restore `CMAKE_REQUIRED_FLAGS` before and after modifying it
   `set(OLD_CMAKE_REQUIRED_FLAGS ${CMAKE_REQUIRED_FLAGS})`
   ...
   `set(CMAKE_REQUIRED_FLAGS ${OLD_CMAKE_REQUIRED_FLAGS})`
3. they do single line
   `return static_cast<int>(std::atomic<int64_t>{});`
   so, no porential `unused variable` warnings that may affect compilation results.